### PR TITLE
Invite by email a new user failed (#479)

### PR DESCRIPTION
**Pull request by phlniji**
#479

### CODE REVIEW

File: Tchap/Managers/ThirdPartyIDResolver/ThirdPartyIDResolver.swift [hunk 1]
```diff
@@ -40,9 +40,9 @@ final class ThirdPartyIDResolver: NSObject, ThirdPartyIDResolverType {
         guard let identityServerURL = URL(string: identityServer) else {
             return nil
         }
-        
+        let accessToken = session.credentials.accessToken
         let identityService = MXIdentityService(identityServer: identityServerURL,
-                                                accessToken: nil,
+                                                accessToken: accessToken,
```
Comment (giomfo) [inline Tchap/Managers/ThirdPartyIDResolver/ThirdPartyIDResolver.swift:45]: You could not provide this access token here

File: Tchap/Managers/Invite/InviteService.swift [hunk 2]
```diff
@@ -120,7 +111,7 @@ final class InviteService: InviteServiceType {
     // Check whether a Tchap account has been created for this email. The closure returns a nil identifier when no account exists.
     private func discoverUser(with email: String, completion: @escaping (MXResponse<InviteServiceDiscoverUserResult>) -> Void) {
         let pid = MX3PID(medium: .email, address: email)
-        _ = self.identityService.lookup3PIDs([pid]) { response in
+        _ = self.session.identityService.lookup3PIDs([pid]) { response in
```
Comment (giomfo) [inline Tchap/Managers/Invite/InviteService.swift:114]: Your current implementation doesn't satisfy "The closure returns a nil identifier when no account exists."
We will sync about in direct

**Review by giomfo — CHANGES_REQUESTED** [review]

File: Tchap/Managers/ThirdPartyIDResolver/ThirdPartyIDResolver.swift (hunk 1 above)
Comment (phlniji) [reply Tchap/Managers/ThirdPartyIDResolver/ThirdPartyIDResolver.swift]: IdentityService instance has been removed here.

File: Tchap/Managers/Invite/InviteService.swift (hunk 2 above)
Comment (phlniji) [reply Tchap/Managers/Invite/InviteService.swift]: This is fixed in the last commit.

File: Tchap/Managers/Invite/InviteService.swift [hunk 3]
```diff
@@ -120,12 +111,16 @@ final class InviteService: InviteServiceType {
     // Check whether a Tchap account has been created for this email. The closure returns a nil identifier when no account exists.
     private func discoverUser(with email: String, completion: @escaping (MXResponse<InviteServiceDiscoverUserResult>) -> Void) {
         let pid = MX3PID(medium: .email, address: email)
-        _ = self.identityService.lookup3PIDs([pid]) { response in
+        guard let identityService = self.session.identityService else {
```
Comment (giomfo) [inline Tchap/Managers/Invite/InviteService.swift:114]: as a good practice, we would move the guard let as the first position of the block
(let pid... should be archived after the guard let)

**Review by giomfo — APPROVED** [review]
LGTM - test ok

File: Tchap/Managers/Invite/InviteService.swift (hunk 3 above)
Comment (phlniji) [reply Tchap/Managers/Invite/InviteService.swift:113]: I moved this on the last commit.

**Review by giomfo — APPROVED** [review]

**Pull request merged**